ui/dashboards/src/components/Panel/Panel.tsx

```diff
@@ -91,6 +135,9 @@ export const Panel = memo(function Panel(props: PanelProps) {
 
   const { queryResults } = useDataQueriesContext();
 
+  // Extract project name from URL - memoized to prevent unnecessary re-calculations
+  const projectName = useMemo(() => extractProjectNameFromUrl(), []);
```

Comment:
I'm not sure this is the best way, we are making the assumption that this panel would be used on a specific URL pattern. Maybe would be better to pas this as an optional prop.